proposals/extension-indexers.md

```diff
@@ -4,18 +4,210 @@
 
 ## Declaration
 
-Like all extension members, extension indexers are declared within an extension block:
+### Grammar
 
-``` c#
-public static class Indexers
+Extension indexers are added to the set of permitted members inside an extension
+declaration by extending the grammar as follows (relative to
+[proposals/csharp-14.0/extensions.md](proposals/csharp-14.0/extensions.md#declaration)):
+
+```antlr
+extension_member_declaration
+        : method_declaration
+        | property_declaration
+        | indexer_declaration // new
+        | operator_declaration
+        ;
+```
+
+Like ordinary indexers, extension indexers have no identifier and are identified
+by their parameter list. Extension indexers may use the full set of features that
+ordinary indexers support today (accessor bodies, expression-bodied members,
+ref-returning accessors, `scoped` parameters, attributes, etc.).
+
+Because indexers are always instance members, an extension block that declares
+an indexer must provide a named receiver parameter.  
+
+The existing restrictions on extension members continue to apply: indexers inside an
+extension declaration cannot specify `abstract`, `virtual`, `override`, `new`,
+`sealed`, `partial`, `protected` (or any of the related accessibility modifiers),
+or `init` accessors.
+
+```csharp
+public static class BitExtensions
 {
     extension(int i)
     {
-        public bool this[int bit] => ...;
+        public bool this[int index]
+        {
+            get => ...;
+        }
     }
 }
 ```
 
-Extension indexers declarations generally follow the rules for 
-non-extension [indexers in the Standard](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/classes.md#159-indexers).
+All rules from the C# standard that apply to ordinary indexers apply to extension indexers,
+but extension members do not have an implicit or explicit `this`.
+
+### `IndexerName` attribute
+
+`IndexerNameAttribute` may be applied to an extension indexer. The attribute is
+carried over to the extension member in metadata and its value determines
```

Comment:
This feels somewhat misleading. It looks like the attribute isn't actually emitted, but "carried over" can be interpreted as it is.